nexus/src/external_api/console_api.rs

```diff
@@ -387,17 +401,34 @@ pub struct LoginPathParam {
     pub silo_name: crate::db::model::Name,
 }
 
+#[endpoint {
+   method = GET,
+   path = "/login/{silo_name}/local",
+   tags = ["login"],
+   unpublished = true,
+}]
+pub async fn login_local_begin(
+    rqctx: RequestContext<Arc<ServerContext>>,
+    _path_params: Path<LoginPathParam>,
+) -> Result<Response<Body>, HttpError> {
+    // TODO: figure out why instrumenting doesn't work
+    // let apictx = rqctx.context();
+    // let handler = async { serve_console_index(rqctx.context()).await };
+    // apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
+    serve_console_index(rqctx.context()).await
+}
+
```

Comment:
I'm not sure this endpoint will need to exist after we have silo in the domain. I was thinking a single `GET /login` endpoint could show either username/password login or a link to the IdP as appropriate depending on the silo. Because this is all done through the console, the generic console page be served as usual, and then the console would have to make an API call to determine which thing to show.

Reply:
It'd be nice if we could avoid having to make an extra API call by including the silo in the header or something. Not for this PR, but in whatever solution we ultimately end up with.

Reply:
Yeah, it really would. I think we might soon get to the point where doing that is less ridiculous than the stuff we're doing to get around it. On the other hand, the total decoupling of the frontend from the API has been very beneficial to my sanity so far.